content/blog/aws-eks-managed-nodes-fargate/index.md

```diff
@@ -0,0 +1,310 @@
+                name: "aws"
+                user: {
+                    exec: {
+                        apiVersion: "client.authentication.k8s.io/v1alpha1"
+                        args      : [
+                            [0]: "token"
+                            [1]: "-i"
+                            [2]: "cluster-eksCluster-560236b"
+                        ]
+                        command   : "aws-iam-authenticator"
+                    }
+                }
+            }
+        ]
+    }
+
+Resources:
+    + 37 created
+
+Duration: 12m12s
+```
+
+A fully functioning production cluster typically requires many other considerations. For those, we've put together [a set of playbooks as part of Pulumi Crosswalk for Kubernetes](/docs/guides/crosswalk/kubernetes) that walk through how to go to production with EKS specifically, in addition to other managed Kubernetes offerings.
+
+Notice, for instance, that we didn't need to even provision any worker nodes. This is thanks to the `eks.Cluster` abstraction creating a default node pool for us, which has an auto-scaling policy that attempts to maintain the `desiredCapacity` while remaining within the bounds of `minSize` and `maxSize`. Sometimes you need more explicit control over the worker nodes, however, for reasons such as having more control over capacity, specializing the kind of compute or storage your workload needs, and so on. This ultimately devolves into managing EC2 instances by hand, which [the `eks.NodeGroup` class supports](https://www.pulumi.com/docs/guides/crosswalk/kubernetes/worker-nodes/).
+
+However, needing to manage individual EC2 instances, scaling policies, and connecting them to the right IAM and networking services, can be tedious and difficult to get right. That's where the new AWS features come into play.
+
+## Using EKS Managed Node Groups
+
+The new EKS feature [Managed Node Groups](https://aws.amazon.com/blogs/containers/eks-managed-node-groups/) simplifies the task of managing explicit pools of worker nodes, at the cost of some amount of control. The `@pulumi/eks` package already had many of these conveniences built-in but this is now an official feature of the AWS platform.
+
+Managed Node Groups will automatically scale the EC2 instances powering your cluster using an Auto Scaling Group (ASG) managed by EKS. This ASG also runs the latest Amazon EKS-optimized Amazon Linux 2 AMI. This is great on one hand &mdash; because updates will be applied automatically for you &mdash; but if you want control over this you will want to manage your own node groups. Finally, security groups, IAM roles, and connecting them together is handled for you.
+
+To opt-in to using Managed Node Groups, you can use the raw [`aws.eks.NodeGroup`](https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/aws/eks/#NodeGroup) class. However, if you're using the `eks.Cluster` abstraction, we've made it easy to use them, with the addition of a new `createManagedNodeGroup` function.
```

Comment:
Though you can create `aws.eks.NodeGroup` and link it to the cluster, the stack will error out because AWS will automatically create the `aws-auth` IAM configmap for us and we get a dupe error when trying to create the pulumi managed configmap.

This creates a race between AWS creating the configmap with only IAM data of the first nodegroup to join, and not any user provided role or user IAM mappings, or other managed nodegroup IAM data for that matter.

To get around this, in `createManagedNodeGroup` we create the configmap first, and then have the managed nodegroup depend on the object to ensure we always create the aws-auth as the user expects, and not AWS.

I'd simplify this statement to only focusing on `createManagedNodeGroup` and not the raw `aws.eks.NodeGroup` class.

Reply:
Ah, good to know.

So, the tension here is, we really should be telling people that the raw building blocks are available. Otherwise, we look like we're getting in the way. We want everybody to know that every low-level building block feature of AWS is exposed by our SDK.

And then we layer on top simplifications. Like this one.

How does this sound: I will reword it to make it clear that if you're using the low level building blocks, you'd use `aws.eks.NodeGroup` directly -- but, for the reason you cite, you should use `createManagedNodeGroup` if you're using our higher level EKS package?

Reply:
> How does this sound: I will reword it to make it clear that if you're using the low level building blocks, you'd use aws.eks.NodeGroup directly -- but, for the reason you cite, you should use createManagedNodeGroup if you're using our higher level EKS package

SGTM 👍